src/sql-parser/src/ast/defs/statement.rs

```diff
@@ -1760,3 +1762,14 @@ impl AstDisplay for DeallocateStatement {
     }
 }
 impl_display!(DeallocateStatement);
+
+/// `RAISE ...`
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct RaiseWarningStatement {}
```

Comment:
Empty structs don't need a "body":

```diff
diff --git a/src/sql-parser/src/ast/defs/statement.rs b/src/sql-parser/src/ast/defs/statement.rs
index 2d6cbbb84..32f345c61 100644
--- a/src/sql-parser/src/ast/defs/statement.rs
+++ b/src/sql-parser/src/ast/defs/statement.rs
@@ -1765,7 +1765,7 @@ impl_display!(DeallocateStatement);
 
 /// `RAISE ...`
 #[derive(Debug, Clone, PartialEq, Eq, Hash)]
-pub struct RaiseWarningStatement {}
+pub struct RaiseWarningStatement;
 
 impl AstDisplay for RaiseWarningStatement {
     fn fmt<W: fmt::Write>(&self, f: &mut AstFormatter<W>) {
diff --git a/src/sql-parser/src/parser.rs b/src/sql-parser/src/parser.rs
index 301eb145a..b5b810f09 100644
--- a/src/sql-parser/src/parser.rs
+++ b/src/sql-parser/src/parser.rs
@@ -4479,7 +4479,7 @@ impl<'a> Parser<'a> {
     /// has already been consumed.
     fn parse_raise(&mut self) -> Result<Statement<Raw>, ParserError> {
         self.expect_keyword(WARNING)?;
-        Ok(Statement::RaiseWarning(RaiseWarningStatement {}))
+        Ok(Statement::RaiseWarning(RaiseWarningStatement))
     }
 }
```

Reply:
IIUC this will become `Statement::Raise` with a `severity: Severity` field?

Reply:
Almost certainly yes, because I don't want to write five different statements, I just did it this way to get this draft out sooner.

Reply:
Unless having five different statements is better! I'm going to have to use (yet another) custom severity enum here, since only the five values mentioned in the PR are accepted by the Postgres equivalent. That means no Error, Fatal or Panic, which seems fine to me, since right now this statement doesn't not have error semantics (e.g it doesn't abort open transactions).

Reply:
Ah, I misunderstood to what degree this is a WIP and that the severity fields were definitely going in before this was ready for a more thorough review. Apologies.

Reply:
Sorry, I should have made that clearer!

Reply:
Done!